Calling `parse_fp()` in v0.5.1 results in "AttributeError: '_thread._local' object has no attribute 'thrift_cache'"

After installing thriftpy2 0.5.1, the following traceback was encountered when calling `parse_fp()`:
```
...
 /opt/conda/envs/test/lib/python3.9/site-packages/cugraph_service_client/cugraph_service_thrift.py:262: in <module>
    spec = thriftpy2.load_fp(io.StringIO(cugraph_thrift_spec), module_name="cugraph_thrift")
/opt/conda/envs/test/lib/python3.9/site-packages/thriftpy2/parser/__init__.py:171: in load_fp
    thrift = parse_fp(source, module_name)
/opt/conda/envs/test/lib/python3.9/site-packages/thriftpy2/parser/parser.py:628: in parse_fp
    if enable_cache and module_name in threadlocal.thrift_cache:
E   AttributeError: '_thread._local' object has no attribute 'thrift_cache'
```

Perhaps the change to `parse()` done in [this PR](https://github.com/Thriftpy/thriftpy2/pull/267/files) should also be applied to `parse_fp()`?

--- a/tests/test_parser.py
+++ b/tests/test_parser.py
@@ -3,6 +3,7 @@
 import threading
 
 import pytest
+
 from thriftpy2.thrift import TType
 from thriftpy2.parser import load, load_fp
 from thriftpy2.parser.exc import ThriftParserError, ThriftGrammarError
@@ -43,10 +44,37 @@ def test_cpp_include():
     load('parser-cases/cpp_include.thrift')
 
 
-def test_load_in_sub_thread(reraise):
+@pytest.fixture
+def reset_parser_threadlocal():
+    def delattr_no_error(o, name):
+        try:
+            delattr(o, name)
+        except AttributeError:
+            pass
+
+    from thriftpy2.parser.parser import threadlocal
+    delattr_no_error(threadlocal, 'thrift_stack')
+    delattr_no_error(threadlocal, 'include_dirs_')
+    delattr_no_error(threadlocal, 'thrift_cache')
+    delattr_no_error(threadlocal, 'incomplete_type')
+    delattr_no_error(threadlocal, 'initialized')
+
+
+def test_load_in_sub_thread(reraise, reset_parser_threadlocal):
     @reraise.wrap
     def f():
         load('addressbook.thrift')
+
+    t = threading.Thread(target=f)
+    t.start()
+    t.join()
+
+
+def test_load_fp_in_sub_thread(reraise, reset_parser_threadlocal):
+    @reraise.wrap
+    def f():
+        with open('container.thrift') as fp:
+            load_fp(fp, 'container_thrift')
 
     t = threading.Thread(target=f)
     t.start()

--- a/thriftpy2/parser/parser.py
+++ b/thriftpy2/parser/parser.py
@@ -621,6 +621,15 @@ def parse_fp(source, module_name, lexer=None, parser=None, enable_cache=True):
     :param enable_cache: if this is set to be `True`, parsed module will be
                          cached by `module_name`, this is enabled by default.
     """
+    # threadlocal should be initialized in every threads
+    initialized = getattr(threadlocal, 'initialized', None)
+    if initialized is None:
+        threadlocal.thrift_stack = []
+        threadlocal.include_dirs_ = ['.']
+        threadlocal.thrift_cache = {}
+        threadlocal.incomplete_type = CurrentIncompleteType()
+        threadlocal.initialized = True
+
     if not module_name.endswith('_thrift'):
         raise ThriftParserError('thriftpy2 can only generate module with '
                                 '\'_thrift\' suffix')

